File: 11-20/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    let length = height.length;
    let i = 0,
        j = length - 1;
    let maxArea = 0;
    while (i < j) {
        maxArea = Math.max(
            maxArea,
            Math.min(height[i], height[j]) * (j - i)
        );
        if (height[i] <= height[j]) {
            i++;
        } else {
            j--;
        }
    }
    return maxArea;
};

File: 11-20/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxVal = 0
        i,j =0,len(height)-1

        while i<j:
            area = (j-i)*min(height[i],height[j])
            maxVal = max(maxVal, area)
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
        
        return maxVal